src/createEventEmitterProxy.ts

```diff
@@ -10,6 +10,20 @@ type EventEmitterLike = {
   // eslint-disable-next-line @typescript-eslint/ban-types
   rawListeners(eventName: string | symbol): Function[];
   removeAllListeners(event?: string | symbol): EventEmitterLike;
+  on(name: string, handler: (...args: unknown[]) => unknown): void;
+  prependListener(name: string, handler: (...args: unknown[]) => unknown): void;
+  addListener(name: string, handler: (...args: unknown[]) => unknown): void;
+  off(name: string, handler: (...args: unknown[]) => unknown): void;
+  once(name: string, handler: (...args: unknown[]) => unknown): void;
+};
+
+type EventDetails = {
+  // the name of the event
+  name: string;
+  // the method given to handle the event
+  handler: (...args: unknown[]) => unknown;
+  // name of the method used to add the event (on or prepend)
+  addedWith: 'on' | 'prependListener' | 'addListener' | 'once';
```

Comment:
Nit: Can we use JSDoc for these so they'll show up on hover?

``` suggestion
  /**
   * The name of the event.
   */
  name: string;
  /**
   * The method given to handle the event.
   */
  handler: (...args: unknown[]) => unknown;
  /**
   * Name of the method used to add the event.
   */
  addedWith: 'on' | 'prependListener' | 'addListener' | 'once';

Reply:
hmm it should be the same thing no?